### PR TITLE
24 june

### PR DESCRIPTION
**Pull request by shayaanmiyy**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -54,8 +54,9 @@ <h1>JavaScript Sprint</h1>
             <br>
             <button onclick="findRepeatLetter()">Find Repeat Letters</button>
             <button onclick="sortArray()">Sort Array</button>
+            <button onclick="bubbleSort()">Bubble Sort</button>
```
Comment (shayaanmiyy) [inline index.html:57]: Heelo

**Review by shayaanmiyy — COMMENTED** [review]
Please Have a look

File: index.html (hunk 1 above)
Comment (shayaanmiyy) [reply index.html:57]: Resolved

File: arrays.js [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+  logger(["input Array", givenData, "sumOfThreeElements", answer]);
+}
+
+// Write a JavaScript program to rotate the elements left in a given array of integers of length 3.
+function rotateArrayLeft() {
+  const givenData = [1, 2, 4];
+  if (givenData.length == 3) {
+    givenData.push(givenData.shift());
+  }
+  logger(["Input Array", [1, 2, 4], "rotateArrayLeft", givenData]);
```
Comment (niteshsyngenta11) [inline arrays.js:26]: [1, 2, 4] can be replaced with the variable givenData.

File: arrays.js [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+  let answer = false; // Default Value
+  if (givenData[0] === 1 || givenData[givenData.length] === 1) {
+    answer = true;
+  }
+  logger(["input Array", givenData, "lastOrFirst", answer]);
+}
+
+// Write a JavaScript program to reverse the elements of a given array of integers of length 3.
+function reverseArray() {
+  const array = [1, 2, 4];
```
Comment (niteshsyngenta11) [inline arrays.js:41]: We should change the name of the variable.

File: arrays.js [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+  let answer;
+  answer = array.reverse();
+  logger(["input Array", array, "reverseArray", answer]);
+}
+
+// Write a JavaScript program to create an array by taking the first and last elements from a given array of integers. The length must be larger than or equal to 1.
+function createByFirstAndLast() {
+  const array = [1, 2, 4];
```
Comment (niteshsyngenta11) [inline arrays.js:49]: We should change the name of the variable.

File: arrays.js [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+    "input Array",
+    givenStrings,
+    "findLongestString",
+    `The Longest String is ${stringName} which has length of ${stringLength}cm`,
+  ]);
+}
+
+// Write a JavaScript program to divide a given array of positive integers into two parts. First element belongs to the first part, second element belongs to the second part, and third element belongs to the first part and so on. Now compute the sum of two parts and store it in an array of size two.
+function divideAndSum() {
+  const arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
```
Comment (niteshsyngenta11) [inline arrays.js:78]: We should change the name of the variable.

File: arrays.js [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+  } else {
+    return false;
+  }
+}
+
+function areSimilar() {
+  // This is the Main Function which then call areArraysSimilar()
+  const array1 = [1, 2, 3]; // Array 1
+  const array2 = [1, 3, 2]; // Array 2
+  logger(["input Array", array1, array1, areArraysSimilar(array1, array2)]); // true
```
Comment (niteshsyngenta11) [inline arrays.js:120]: We should change the name of the variable.

File: arrays.js [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+
+  const array7 = [1, 2, 4, 3];
+  const array8 = [2, 1, 3, 4];
+  logger(["input Array", array7, array8, areArraysSimilar(array7, array8)]); // false
```
Comment (niteshsyngenta11) [inline arrays.js:124]: We should change the name of the variable.

File: arrays.js [hunk 8]
```diff
@@ -0,0 +1,276 @@
...
+    numbers[index] = 1;
+  });
+  logger(["input Array", numbers, "replaceNumber", numbers]);
+}
+
+// Write a JavaScript program to check if there is at least one element in two given sorted arrays of integers.
+function findElement() {
+  const arr1 = [1, 2, 3, 4];
+  const arr2 = [1, 2, 3, 4, 5, 6, 7, 8, 9];
+  let correctCount = 0;
```
Comment (niteshsyngenta11) [inline arrays.js:140]: We should change the name of the variable.

File: arrays.js [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+  if (correctCount == arr1.length) {
+    answer = !answer;
+  }
+  logger(["input Array", arr1, arr2, "findElement", answer]);
+}
+
+// Write a JavaScript program to find the number of inversions of a given array of integers. Note: Two elements of the array a stored at positions i and j form an inversion if a[i] > a[j] and i < j.
+
+function countInversions() {
+  const arr = [1, 20, 6, 4, 5];
```
Comment (niteshsyngenta11) [inline arrays.js:156]: We should change the name of the variable.

File: arrays.js [hunk 10]
```diff
@@ -0,0 +1,276 @@
...
+  let invCount = 0;
+  const arlen = arr.length;
+  for (let i = 0; i < arlen; i++) {
+    // go on each elem of arr
+    for (let j = i + 1; j < arlen; j++) {
+      // again loop
+      if (arr[i] > arr[j]) {
+        // compae first elem with same arr 2nd elem
```
Comment (niteshsyngenta11) [inline arrays.js:164]: Spelling issues in the comment.

File: arrays.js [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+    }
+  }
+  logger(["input Array", arr, "countInversions", invCount]);
+}
+
+/* Write a JavaScript program to create an array of prefix sums of the given array.
+In computer science, the prefix sum, cumulative sum, inclusive scan, or simply scan of a sequence of numbers x0, x1, x2, ... is a second sequence of numbers y0, y1, y2, ..., the sums of prefixes of the input sequence:y0 = x0y1 = x0 + x1y2 = x0 + x1+ x2 */
+
+function addArrSum() {
+  const arr = [1, 2, 3, 4, 5];
```
Comment (niteshsyngenta11) [inline arrays.js:176]: We should change the name of the variable.

File: arrays.js [hunk 12]
```diff
@@ -0,0 +1,276 @@
...
+
+  return result;
+}
+
+function reduceArrr() {
+  const givenData = [1, 2, 3, 4];
+
+  let sumResult = reduceArray(givenData, 0);
+  let productResult = reduceArray(givenData, 1);
+  let subtractResult = reduceArray(givenData, 10);
```
Comment (niteshsyngenta11) [inline arrays.js:235]: Can these be const instead of let.

File: operator-expression.js [hunk 13]
```diff
@@ -0,0 +1,225 @@
...
+                        <label for="itemPrice" class="form-label">Item Price</label>
+                        <input type="number" required name="itemPrice${counter}" class="form-control" id="itemPrice${counter}">
+                    </div>
+                    <div class="mb-3">
+                        <label for="itemQuantity${counter}" class="form-label">Item Quantity</label>
+                        <input type="number" required name="itemQuantity${counter}" class="form-control" id="itemQuantity${counter}">
+                    </div>
+                </div>`;
+  document
+    .getElementById("inputForm")
```
Comment (niteshsyngenta11) [inline operator-expression.js:84]: We should not use generic ids. It should be more specific.

File: operator-expression.js [hunk 14]
```diff
@@ -0,0 +1,225 @@
...
+                        <label for="itemName${counter}" class="form-label">Item Name</label>
+                        <input type="text" required name="itemName${counter}" class="form-control" id="itemName${counter}">
+                    </div>
+                    <div class="mb-3">
+                        <label for="itemPrice" class="form-label">Item Price</label>
+                        <input type="number" required name="itemPrice${counter}" class="form-control" id="itemPrice${counter}">
+                    </div>
+                    <div class="mb-3">
+                        <label for="itemQuantity${counter}" class="form-label">Item Quantity</label>
+                        <input type="number" required name="itemQuantity${counter}" class="form-control" id="itemQuantity${counter}">
```
Comment (niteshsyngenta11) [inline operator-expression.js:80]: Do we need IDs?

File: operator-expression.js [hunk 15]
```diff
@@ -0,0 +1,225 @@
...
+
+// Create a program to calculate the monthly salary of an employee. Prompt the user to enter the hourly wage and the number of hours worked in a month. Use the multiplication operator to calculate the monthly salary by multiplying the hourly wage and the number of hours worked.
+
+function employeeSalary() {
+  const salary = prompt("What is your Hourly Salary?"); // Prompt User
+  const workinghour = prompt("How long did your worked in a month? (in hours)"); // Prompt User
+  let answer;
+  if (Number(workinghour) && Number(salary)) {
+    // Check for Numbers
+    answer = `Your Montly Salary for the time you wokred will be ${
```
Comment (niteshsyngenta11) [inline operator-expression.js:96]: Fix the typo.

File: operator-expression.js [hunk 16]
```diff
@@ -0,0 +1,225 @@
...
+
+// Write a JavaScript program to check whether it is possible to replace $ in a given expression x $ y = z with one of the four signs +, -, * or / to obtain a correct expression.   For example x = 10, y = 30 and z = 300, we can replace $ with a multiple operator (*) to obtain x * y = z
+
+function operatorCheck() {
+  const x = 10;
+  const y = 30;
+  const z = 300; // 40, -20,
+  const ops = ["+", "-", "/", "*"];
+  let answer = "n/a";
+  ops.forEach((o) => {
```
Comment (niteshsyngenta11) [inline operator-expression.js:164]: Use proper variable name

**Review by niteshsyngenta11 — COMMENTED** [review]
General feedback:
- Commenting should be proper
- Use proper variable names
- Use const instead of let wherever applicable.

File: arrays.js (hunk 3 above)
Comment (shayaanmiyy) [reply arrays.js:41]: Yes, I changed that on pull request 3, That is Yesterday

File: arrays.js (hunk 4 above)
Comment (shayaanmiyy) [reply arrays.js:49]: Yes, I changed that on pull request 3, That is Yesterday

File: arrays.js (hunk 5 above)
Comment (shayaanmiyy) [reply arrays.js:78]: Yes, I changed that on pull request 3, That is Yesterday

File: arrays.js (hunk 6 above)
Comment (shayaanmiyy) [reply arrays.js:120]: Yes, I changed that on pull request 3, That is Yesterday

**Pull request merged**